components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,358 @@
+
+        this.uri = new URI(this.uriString);
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+
+        ReflectionUtils.setFieldValue(this.patientResource, "uriInfo", this.uriInfo);
+    }
+
+    //----------------------------Get Patient Tests----------------------------
+
+    @Test
+    public void getPatientCantFindPatient() throws XWikiRestException {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("No such patient record: [{}]", this.id);
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientUserDoesNotHaveAccess() throws XWikiRestException {
+        doReturn(false).when(this.access).hasAccess(Right.VIEW, null, null);
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("View access denied to user [{}] on patient record [{}]", this.currentUser, this.id);
+        Assert.assertEquals(Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void getPatientAddsLinksToPatientJSON() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, null, null);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        Assert.assertTrue(response.getEntity() instanceof JSONObject);
+
+        Map<String, Map<String, String>> json = (Map)response.getEntity();
+        Assert.assertThat(json, hasValue(hasEntry("rel", Relations.SELF)));
+        Assert.assertThat(json, hasValue(hasEntry("href", this.uriString)));
+        Map<String, List<MediaType>> actualMap = (Map)response.getMetadata();
+        Assert.assertThat(actualMap, hasValue(hasItem(MediaType.APPLICATION_JSON_TYPE)));
+    }
+
+    @Test
+    public void getPatientAlwaysSendsLoggerMessageOnRequest() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, null, null);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
+
+        this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("Retrieving patient record [{}] via REST", this.id);
+    }
+
+    //----------------------------Update Patient Tests----------------------------
+
+    @Test
+    public void updatePatientCantFindPatient() throws XWikiRestException {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+        WebApplicationException ex = null;
+        try {
+            this.patientResource.updatePatient("", this.id);
+        } catch (WebApplicationException temp) {
+            ex = temp;
+        }
+        Assert.assertNotNull("updatePatient did not throw a WebApplicationException as expected " +
+            "when the patient could not be found", ex);
```

Comment:
I wanted to test that the `WebApplicationException` was thrown here, but also check the status code passed to the exception and verify the logger message, because It seemed like was all essentially the same behaviour.  I started by using the `@Test(expected=WebApplicationException.class)` notation, but it seems that the test stops executing when the exception is thrown, making testing for the status code and the logger difficult